code/socialDistribution/views.py

```diff
@@ -300,34 +300,65 @@ def author(request, author_id):
 def create(request):
     return render(request, 'create/index.html')
 
+def post(request, author_id, post_id):
+    """
+        Allows user to view a specific post.
+        Mostly useful to share public (unlisted) posts through url
+    """
+    author = get_object_or_404('Author', pk=author_id)
+    post = get_object_or_404('Post', pk=post_id)
+    pass
+
 
 def posts(request, author_id):
     """ Handles POST request to publish a post.
 
         Allows user to create a post. The newly created post will also be rendered. 
     """
-    author = get_object_or_404(LocalAuthor, pk=author_id)
     user_id = LocalAuthor.objects.get(user=request.user).id
 
     if request.method == 'POST':
         form = PostForm(request.POST, request.FILES, user_id=user_id)
         if form.is_valid():
-            bin_content = form.cleaned_data.get('content_media')
-            if bin_content is not None:
-                content_media = base64.b64encode(bin_content.read())
+            media = form.cleaned_data.get('content_media')
+            content = form.cleaned_data.get('content_text')
+            if media is not None:
+                uploaded_img = Image.objects.create(
+                    caption=form.cleaned_data.get('description'),
+                    image=media
+                )
```

Comment:
Can we store both content_media or content_text (which ever is filled in) in a `content` `BinaryField`? I talked to Michelle about this but I can send you what I sent her.